sdk/messaging/azservicebus/receiver_test.go

```diff
@@ -19,6 +19,21 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestReceiverCancel(t *testing.T) {
+	serviceBusClient, cleanup, queueName := setupLiveTest(t, nil)
+	defer cleanup()
+
+	receiver, err := serviceBusClient.NewReceiverForQueue(queueName, nil)
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	messages, err := receiver.ReceiveMessages(ctx, 5, nil)
+	require.NoError(t, err)
```

Comment:
Maybe I'm missing something here but why is `err` expected to be `nil`?  Hasn't the context timed out?

Reply:
Oops, sorry, need to fix that one. I was focusing on the unit tests, but yes, the cancel error should come back here.

Reply:
(this was the previous undesirable behavior!)

Reply:
(fixed, btw)